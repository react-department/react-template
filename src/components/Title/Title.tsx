import { testIdMainTitle } from '../../constants/TestId';

import type { ReactElement } from 'react';
import type ITitle from './interfaces/ITitle';

import styles from './sass/Title.module.scss';

/**
 * H1 title component
 */
export default function Title({ text }: ITitle): ReactElement {
  return (
    <h1 className={styles.appTitle} data-testid={testIdMainTitle}>
      {text}
    </h1>
  );
}
