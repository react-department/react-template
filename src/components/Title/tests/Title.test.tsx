import { screen } from '@testing-library/react';

import { render } from '../../../../__tests__/test-utils';
import Title from '../Title';

describe('<Title />', () => {
  test('should render component', () => {
    const { asFragment } = render(<Title text="Title" />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('should render component with title', () => {
    render(<Title text="Title" />);
    expect(screen.getByText('Title')).toBeInTheDocument();
  });
});
