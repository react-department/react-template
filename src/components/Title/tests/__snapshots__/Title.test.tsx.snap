// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render component 1`] = `
<DocumentFragment>
  <h1
    class="appTitle"
    data-testid="mainTitle"
  >
    Title
  </h1>
</DocumentFragment>
`;
